packages/projection/test/operators/Mappers/withHandle.test.ts

```diff
@@ -0,0 +1,145 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { Buffer } from 'buffer';
+import { HexBlob } from '@cardano-sdk/util';
+import { ProjectionEvent } from '../../../src';
+import { firstValueFrom, of } from 'rxjs';
+import { withHandles } from '../../../src/operators/Mappers';
+
+const assetIdFromNameLabelAndPolicyId = (assetName: string, policyId: Cardano.PolicyId): Cardano.AssetId =>
+  Asset.util.assetIdFromPolicyAndName(policyId, Cardano.AssetName(Buffer.from(assetName).toString('hex')));
+
+const handlePolicyId = Cardano.PolicyId('f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a');
+const assetIdFromHandle = (handle: string) => assetIdFromNameLabelAndPolicyId(handle, handlePolicyId);
+
+describe('withHandles', () => {
+  it('sets "datum" property on the handle if utxo has datum', async () => {
+    const datum = HexBlob('123abc');
+    const outputs: Cardano.TxOut[] = [
+      {
+        address: Cardano.PaymentAddress(
+          'addr_test1qretqkqqvc4dax3482tpjdazrfl8exey274m3mzch3dv8lu476aeq3kd8q8splpsswcfmv4y370e8r76rc8lnnhte49qqyjmtc'
+        ),
+        datum,
+        value: {
+          assets: new Map([[assetIdFromHandle('somehandle'), 1n]]),
+          coins: 25_485_292n
+        }
+      }
+    ];
+    const validTxSource$ = of({
+      block: { body: [{ body: { outputs } }] }
+    } as ProjectionEvent);
+    const { handles } = await firstValueFrom(
+      validTxSource$.pipe(
+        withHandles({
+          policyIds: [handlePolicyId]
+        })
+      )
+    );
+
+    expect(handles[0].datum).toEqual(datum);
+  });
+
+  it('maps and filters assets, from outputs, containing handles matching the given policy ID to an array of objects', async () => {
```

Comment:
Need 1 more test (and probably update the implementation to make it pass)
```suggestion
  describe('multiple transactions in 1 block minting/transferring the same handle', () => {
    it('reduces handles to a single (latest) address');
  });
  it('maps and filters assets, from outputs, containing handles matching the given policy ID to an array of objects', async () => {
```

Reply:
This was added on e7b66de

Reply:
@greatertomi can you link to the lines of code where these tests are implemented?

Reply:
Fixed on aa3d555

Reply:
This looks good to me. Just need final approval from @mkazlauskas